From 8bc8cef9aee668f9e14000e8450cd55d75ebc473 Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert+renesas@glider.be>
Date: Fri, 20 Nov 2015 14:47:50 +0100
Subject: [PATCH] kbuild: Enable DT symbols when CONFIG_OF_OVERLAY is used

commit de9960a87ccdb63d790d7b93eb2425cd3a5e107e from linux-socfpga

Signed-off-by: Geert Uytterhoeven <geert+renesas@glider.be>
[OP: backport of de9960a87ccd from socfpga-4.18 branch of linux-socfpga]
Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 scripts/Makefile.lib | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 1bb594fcfe12..32d68c153019 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -282,6 +282,10 @@ cmd_dt_S_dtb=						\
 $(obj)/%.dtb.S: $(obj)/%.dtb FORCE
 	$(call if_changed,dt_S_dtb)
 
+ifeq ($(CONFIG_OF_OVERLAY),y)
+DTC_FLAGS += -@
+endif
+
 quiet_cmd_dtc = DTC     $@
 cmd_dtc = mkdir -p $(dir ${dtc-tmp}) ; \
 	$(CPP) $(dtc_cpp_flags) -x assembler-with-cpp -o $(dtc-tmp) $< ; \
-- 
2.17.1

